apps/docs/utils/parse-markdown.tsx

```
@@ -0,0 +1,256 @@
+ * everything on the page.
+ */
+export function parseHeadings(content: string): {
+	headings: ParsedHeading[]
+	initialContentText: string
+} {
+	const headings: Omit<ParsedHeading, 'slug'>[] = [
+		{
+			level: 0,
+			title: '',
+			contentText: '',
+			isInherited: false,
+		},
+	]
+
+	remark()
+		.use(remarkMdx)
+		.use(() => (tree: Root) => {
+			interface State {
+				isInHeading: boolean
+				isInherited: boolean
+			}
+
+			function visitChildren(node: Nodes, state: State) {
+				if ('children' in node) {
+					for (const child of node.children) {
+						visit(child, state)
+					}
+				}
+			}
+
+			function heading() {
+				const currentHeading = headings[headings.length - 1]
+				return currentHeading
+			}
+
+			function visit(node: Nodes, state: State) {
+				switch (node.type) {
+					case 'heading': {
+						const { depth } = node
+
+						headings.push({
+							level: depth,
+							title: '',
+							contentText: '',
+							isInherited: state.isInherited,
+						})
+
+						visitChildren(node, { ...state, isInHeading: true })
+
+						break
+					}
+
+					case 'text':
+					case 'inlineCode':
+						if (state.isInHeading) {
+							heading().title += node.value
+						} else {
+							heading().contentText += node.value
+						}
+
+						break
+
+					case 'blockquote':
+					case 'paragraph':
+						assert(!state.isInHeading)
+						if (!heading().contentText.endsWith('\n')) {
+							heading().contentText += '\n'
+						}
+						visitChildren(node, state)
+						break
+
+					case 'code':
+						// skip these entirely
+						break
+
+					case 'mdxJsxFlowElement':
+					case 'mdxJsxTextElement': {
+						const { name, attributes } = node
+
+						if (
+							// Remove table of contents
+							(name === 'details' &&
+								attributes.some(
+									(attr) =>
+										'name' in attr &&
+										attr.name === 'className' &&
+										attr.value === 'article__table-of-contents'
```

it's not the biggest deal but like we talked about this maybe failing silently later, you could pull `article__table-of-contents` out something exported and shared with `getApiMarkdown`'s usage of it